toml/_parser.ts

```diff
@@ -123,6 +123,14 @@ export class Scanner {
   isCurrentCharEOL() {
     return this.char() === "\n" || this.slice(0, 2) === "\r\n";
   }
+
+  match(regExp: RegExp) {
+    const match = this.#source.slice(this.#position).match(regExp);
```

Comment:
`#source.slice(this.#position)` this part creates large substring of source input. This seems defeating the purpose of Scanner class. I think we should keep the existing way of matching against small part of source string.

Reply:
I agree that using `slice()` is not very efficient. I updated the PR so it doesn't slice the string. Could you take another look?